### DISCUSSION TITLE
How to add a tool in Self Hosted Renovate and use it in postUpgradeTasks?

### DISCUSSION THREAD

**Discussion by PJ-Schulz** [renovatebot/renovate#28444] [Request Help]
### What would you like help with?

None

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Kubernetes renovate/renovate:37.281-full

### Please tell us more about your question or problem

I want to execute the tool `yq` in the postUpgradeTasks. There i want to extract the updated version and write it into another file. I am using the full image, but `yq` is not installed.

How can I add this tool, without building a new image?

When i do something like this i get an error, that i can not install anything:

```json
  "postUpgradeTasks": {
    "commands": [
      "apt install yq",
      "yq '.images[] | select(.name == \"modbus-image\") | .newTag' kubernetes/example/kustomization.yaml > tag.txt" ],
    "fileFilters": [
      "kubernetes/example/kustomization.yaml"
    ],
    "executionMode": "branch"
  },
```

How can i install a tool after renovate has started up, and use it in postUpgradeTasks?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
If you don't want to build a new image then you need to find a way to install two as non-root

**Comment by PJ-Schulz** [comment]
using the installation script `install-tool yq 4.30.6` also fails.
Is it possible to start a docker container at this point where yq is installed?

In my config I have set this configuration:

env:
* RENOVATE_BINARY_SOURCE: global

config.json:
* "allowedPostUpgradeCommands": [".*"]

**Reply by rarkins** [reply]
Install-tool does not support yq

Second question does not make sense to me

**Reply by PJ-Schulz** [reply]
I was thinking of the following. I have a Docker image that is built with this Dockerfile, lets call the image `alpine-yq:v1`
```
FROM alpine:3.19
RUN apk add apache2-utils
```

Now I execute something like this as postUpgradeTasks.  `docker run alpine-yq:v1 yq ... > output.txt`

```json
"postUpgradeTasks": {
    "commands": ["docker run alpine-yq:v1 yq ... > output.txt" ],
    "fileFilters": [
      "kubernetes/example/kustomization.yaml",
       ...
    ],
    "executionMode": "branch"
  },
```

Is there any way to do something like this?

**Reply by rarkins** [reply]
You might be able to, but docker-in-docker is another challenge. e.g. you'd need to mount the docker socket into the container
